### PR TITLE
Send attachments in trackException

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Modify the test apps to send attachments even when "handle errors" toggle is on. Also ensure that the attachments functionality remains enabled in this case.

Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Modify the test apps to send attachments even when "handle errors" toggle is on. Also ensure that the attachments functionality remains enabled in this case.

## Related PRs or issues

[AB#71074](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/71074)

### CODE REVIEW

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/crashes/CrashesPrivateHelper.java [hunk 1]
```diff
@@ -5,14 +5,17 @@
...
+
 import java.util.Map;
 
 public final class CrashesPrivateHelper {
 
     private CrashesPrivateHelper() {
     }
 
-    public static void trackException(Throwable throwable, Map<String, String> properties) {
+    public static void trackException(Throwable throwable, Map<String, String> properties, Iterable<ErrorAttachmentLog> attachments) {
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/crashes/CrashesPrivateHelper.java]: This no longer exists on the track exception branch. This feature must be based on `feature/track-exception` branch.

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/ManagedErrorActivity.java [hunk 2]
```diff
@@ -72,7 +77,9 @@ protected void send(View view) {
                 Method method = Crashes.class.getDeclaredMethod("trackException", Throwable.class, Map.class);
                 method.setAccessible(true);
                 method.invoke(null, t, properties);
-
+                AttachmentsUtils.getInstance().setFileAttachment(sSharedPreferences.getString(FILE_ATTACHMENT_KEY, null));
```
Comment (annakocheshkova) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/ManagedErrorActivity.java]: I don't understand the logic here. `AttachmentUtils` is a singleton and stores attachments in memory. 
But they are retrieved from preferences only here? And where are they saved to preferences or deleted from them?
If it's only [in MainActivity](https://github.com/microsoft/appcenter-sdk-android/blob/develop/apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/MainActivity.java#L108), then I'd rather rewrite this. The logic of working with them is all over the place. We don't need a singleton, and `AttachmentUtils` class can work with `SharedPreferences`.

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/ManagedErrorActivity.java [hunk 3]
```diff
@@ -72,7 +74,7 @@ protected void send(View view) {
                 Method method = Crashes.class.getDeclaredMethod("trackException", Throwable.class, Map.class);
                 method.setAccessible(true);
                 method.invoke(null, t, properties);
-
+                Iterable<ErrorAttachmentLog> attachmentLogs = AttachmentsUtil.getInstance().getErrorAttachments(getApplicationContext());
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/ManagedErrorActivity.java]: Now we have the code to send attachments, please merge `feature/handled-error-attachments` to your branch and retarget the PR and test in the app.

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/util/AttachmentsUtil.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
+package com.microsoft.appcenter.sasquatch.util;
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/util/AttachmentsUtil.java]: Copyright missing.

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/util/AttachmentsUtil.java [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    public void setTextAttachment(String textAttachment) {
+        SharedPreferences.Editor editor = sSharedPreferences.edit();
+        if (textAttachment == null) {
+            editor.remove(TEXT_ATTACHMENT_KEY);
+        } else {
+            editor.putString(TEXT_ATTACHMENT_KEY, textAttachment);
+        }
+        editor.apply();
+        this.mTextAttachment = textAttachment;
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/util/AttachmentsUtil.java]: On Android we use `mField` convention, not `this`.

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/util/AttachmentsUtil.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.microsoft.appcenter.sasquatch.activities.MainActivity.LOG_TAG;
+import static com.microsoft.appcenter.sasquatch.activities.MainActivity.sSharedPreferences;
+
+public class AttachmentsUtil {
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/util/AttachmentsUtil.java]: This new file has multiple warnings.

**Comment by guperrot** [comment]
Otherwise works, just fix warnings then good to go.

**Pull request merged**